--- qt-x11-free-3.3.6/mkspecs/linux-g++-32/qmake.conf.xorg7.1	2006-03-08 11:48:20.000000000 -0500
+++ qt-x11-free-3.3.6/mkspecs/linux-g++-32/qmake.conf	2006-09-01 13:14:19.000000000 -0400
@@ -40,12 +40,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_T
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $(QTDIR)/include
 QMAKE_LIBDIR_QT		= $(QTDIR)/lib
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
--- qt-x11-free-3.3.6/mkspecs/linux-g++-64/qmake.conf.xorg7.1	2006-09-01 12:50:46.000000000 -0400
+++ qt-x11-free-3.3.6/mkspecs/linux-g++-64/qmake.conf	2006-09-01 12:50:46.000000000 -0400
@@ -43,12 +43,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_T
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib64
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib64
 QMAKE_INCDIR_QT		= $(QTDIR)/include
 QMAKE_LIBDIR_QT		= $(QTDIR)/lib64
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib64
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib64
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
--- qt-x11-free-3.3.6/mkspecs/linux-g++/qmake.conf.xorg7.1	2006-09-01 12:50:46.000000000 -0400
+++ qt-x11-free-3.3.6/mkspecs/linux-g++/qmake.conf	2006-09-01 13:13:00.000000000 -0400
@@ -40,12 +40,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_T
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $(QTDIR)/include
 QMAKE_LIBDIR_QT		= $(QTDIR)/lib
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
